sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -601,6 +601,264 @@ manager, with `__aenter__`, `__aexit__`, and `close` methods.
+        self._pipeline_response = self._initial_response = initial_response
+        self._deserialization_callback = deserialization_callback
+        self._command = functools.partial(self.client.get_upload_file, file_id=self.file_id, **self._kwargs)
+
+    def status(self) -> str:
+        """Return the current status as a string.
+
+        :rtype: str
+        """
+        return "Succeeded" if self._finished else "InProgress"
+
+    def finished(self) -> bool:
+        """Is this polling finished?
+
+        :rtype: bool
+        """
+        return self._finished
+
+    def resource(self) -> JSON:
+        """Return the built resource."""
+        return self._resource
+
+    def run(self) -> None:
+        """The polling loop"""
+        while not self.finished():
+            try:
+                self._resource = self._command()
+                self._finished = True
+            except ResourceNotFoundError:
+                pass
+            if not self.finished():
+                time.sleep(self._polling_interval)
+
+    def get_continuation_token(self) -> str:
+        import pickle
+
+        return base64.b64encode(pickle.dumps(self._initial_response)).decode("ascii")
+
+    @classmethod
+    def from_continuation_token(cls, continuation_token: str, **kwargs) -> Tuple[Any, Any, Callable]:
+        try:
+            client = kwargs["client"]
+        except KeyError:
+            raise ValueError("Need kwarg 'client' to be recreated from continuation_token")
+
+        import pickle
+
+        initial_response = pickle.loads(base64.b64decode(continuation_token))  # nosec
+        # Restore the transport in the context
+        initial_response.context.transport = client._client._pipeline._transport  # pylint: disable=protected-access
+        return client, initial_response, lambda: None
+```
+
+And now, to plug into the client code:
+
+```python
+class ServiceOperations:
+
+    def begin_upload(self, data: AnyStr, **kwargs) -> LROPoller[JSON]:
+        continuation_token = kwargs.pop("continuation_token", None)
+        polling_method = CustomLROBasePolling(**kwargs)
+        if continuation_token is not None:
+            return LROPoller.from_continuation_token(
+                continuation_token=continuation_token,
+                polling_method=polling_method,
+                client=self
+            )
+        response = self._client.create_upload(data, **kwargs)
+
+        return LROPoller(
+            client=self,
+            initial_response=response,
+            deserialization_callback=lambda: None,
+            polling_method=polling_method,
+        )
+```
+
+Note that we need to account for a `continuation_token` being passed by the user, in which case we should not make the
+initial call again, but rather resume polling from the rehydrated state.
+
+### Poller - LROPoller/AsyncLROPoller
+
+The last option is if you need to customize the public interface of the `LROPoller` / `AsyncLROPoller`.
```

Comment:
Maybe we could add something like:
Reasons to do this might include exposing important attributes or metadata of the operation in progress, or adding new features to interact with the operation, such as to pause/resume or cancel it.